File: pro_kbl_22_kjinsu/sol.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

int IsPrime(unsigned long long n)
{
    unsigned long long c;
    if(n < 2) return 0;
    for (c = 2; c * c <= n; c++)
    {
        if(!(n % c)) return 0;
    }
    return 1;
}
int solution(int n, int k) {
    int answer = 0;
    unsigned long long c,t,r;
    for(r = 1,t = 0; n || t;)
    {
        c = n % k;
        if (!c) {
            answer += IsPrime(t);
            t = 0;
            r= 1;
        } else {
            t += c * r;
            r *= 10;
        }
        n /= k;
    }
    return answer;
}

File: pro_kbl_22_kjinsu/sol.py
def conv(n, k):
    s = ''
    while n:
        s += str(n%k)
        n //= k
    return s[::-1]

# n 이 소수인지 판정 
def isPrime(n):
    if n <= 1: return False
    i = 2
    while i * i <= n:
        if n%i ==0: return False
        i += 1
    return True

def solution(n, k):
    s= conv(n, k)
    cnt = 0
    for num in s.split('0'):
        if not num: continue # 빈 문자열은 예외처디 ><
        if isPrime(int(num)): cnt += 1
    return cnt